deps/patches/http_parser-makefile.patch

```diff
@@ -0,0 +1,112 @@
+@@ -19,11 +19,24 @@
+ # IN THE SOFTWARE.
+ 
+ PLATFORM ?= $(shell sh -c 'uname -s | tr "[A-Z]" "[a-z]"')
++ifeq ($(findstring cygwin,$(PLATFORM)), cygwin)
++PLATFORM = winnt
++endif
++ifeq ($(findstring mingw,$(PLATFORM)), mingw)
++PLATFORM = winnt
++endif
++ifeq ($(findstring msys,$(PLATFORM)), msys)
++PLATFORM = winnt
++endif
++
+ HELPER ?=
+ BINEXT ?=
+ ifeq (darwin,$(PLATFORM))
+ SONAME ?= libhttp_parser.2.7.0.dylib
+ SOEXT ?= dylib
++else ifeq (winnt,$(PLATFORM))
++SONAME ?= libhttp_parser-2.7.dll
++SOEXT  ?= dll
+ else ifeq (wine,$(PLATFORM))
+ CC = winegcc
+ BINEXT = .exe.so
+@@ -50,11 +61,18 @@
+ CFLAGS_DEBUG = $(CFLAGS) -O0 -g $(CFLAGS_DEBUG_EXTRA)
+ CFLAGS_FAST = $(CFLAGS) -O3 $(CFLAGS_FAST_EXTRA)
+ CFLAGS_BENCH = $(CFLAGS_FAST) -Wno-unused-parameter
++
++ifeq (winnt,$(PLATFORM))
++CFLAGS_LIB = $(CFLAGS_FAST)
++else
+ CFLAGS_LIB = $(CFLAGS_FAST) -fPIC
++endif
+ 
+ LDFLAGS_LIB = $(LDFLAGS) -shared
+ 
+ INSTALL ?= install
++STRIP ?= strip
++
+ PREFIX ?= $(DESTDIR)/usr/local
+ LIBDIR = $(PREFIX)/lib
+ INCLUDEDIR = $(PREFIX)/include
+@@ -64,6 +82,8 @@
+ LDFLAGS_LIB += -Wl,-soname=$(SONAME)
+ endif
+ 
++all: library package
++
+ test: test_g test_fast
+ 	$(HELPER) ./test_g$(BINEXT)
+ 	$(HELPER) ./test_fast$(BINEXT)
+@@ -101,10 +121,12 @@
+ libhttp_parser.o: http_parser.c http_parser.h Makefile
+ 	$(CC) $(CPPFLAGS_FAST) $(CFLAGS_LIB) -c http_parser.c -o libhttp_parser.o
+ 
+-library: libhttp_parser.o
++library: $(SONAME)
++$(SONAME): libhttp_parser.o
+ 	$(CC) $(LDFLAGS_LIB) -o $(SONAME) $<
+ 
+-package: http_parser.o
++package: libhttp_parser.a
++libhttp_parser.a: http_parser.o
+ 	$(AR) rcs libhttp_parser.a http_parser.o
+ 
+ url_parser: http_parser.o contrib/url_parser.c
+@@ -122,26 +144,28 @@
+ tags: http_parser.c http_parser.h test.c
+ 	ctags $^
+ 
+-install: library
+-	$(INSTALL) -D  http_parser.h $(INCLUDEDIR)/http_parser.h
+-	$(INSTALL) -D $(SONAME) $(LIBDIR)/$(SONAME)
+-	ln -s $(LIBDIR)/$(SONAME) $(LIBDIR)/libhttp_parser.$(SOEXT)
+-
+-install-strip: library
+-	$(INSTALL) -D  http_parser.h $(INCLUDEDIR)/http_parser.h
+-	$(INSTALL) -D -s $(SONAME) $(LIBDIR)/$(SONAME)
+-	ln -s $(LIBDIR)/$(SONAME) $(LIBDIR)/libhttp_parser.$(SOEXT)
++install: library package
++	$(INSTALL) -d                    $(DESTDIR)$(INCLUDEDIR)
++	$(INSTALL) -m 0644 http_parser.h $(DESTDIR)$(INCLUDEDIR)/http_parser.h
++	$(INSTALL) -d                $(DESTDIR)$(LIBDIR)
++	$(INSTALL) -m 0644 $(SONAME) $(DESTDIR)$(LIBDIR)/$(SONAME)
++	ln -sf             $(SONAME) $(DESTDIR)$(LIBDIR)/libhttp_parser.$(SOEXT)
++	$(INSTALL) -m 0644 libhttp_parser.a $(DESTDIR)$(LIBDIR)/libhttp_parser.a
++
++install-strip: install
++	$(STRIP) $(DESTDIR)$(LIBDIR)/$(SONAME)
+ 
+ uninstall:
+-	rm $(INCLUDEDIR)/http_parser.h
+-	rm $(LIBDIR)/$(SONAME)
+-	rm $(LIBDIR)/libhttp_parser.so
++	rm $(DESTDIR)$(INCLUDEDIR)/http_parser.h
++	rm $(DESTDIR)$(LIBDIR)/$(SONAME)
++	rm $(DEstdiR)$(LIBDIR)/libhttp_parser.$(SOEXT)
++	rm $(DEstdiR)$(LIBDIR)/libhttp_parser.a
```

Comment:
This was part of the original patch in the main http-parser PR, I'll clean this patch up.